### ISSUE TITLE
GraphQL plan request takes ~20 seconds

### ISSUE THREAD

**Issue by trevorgerhardt** [conveyal/r5#175]
Using @buma's example (https://github.com/buma/R5Docs) with [GraphiQL](https://github.com/skevy/graphiql-app) I am getting requests that consistently take around 20 seconds.

Using all modes in the plan request:

``` graphql
directModes: [BICYCLE, BICYCLE_RENT, CAR, WALK], accessModes: [WALK, BICYCLE, BICYCLE_RENT, CAR_PARK], egressModes: [WALK], transitModes: [TRANSIT]
```

Query variables:

``` json
{
  "fromLat": 38.943037,
  "fromLon": -77.029199,
  "toLat": 38.891254,
  "toLon": -77.085016,
  "fromTime": "2016-06-16T07:00:00-05:00",
  "toTime": "2016-06-16T08:00:00-05:00"
}
```

Built and run graphs with r5 version 54c25a566501040433d69c16e370446840be1cbe:

``` shell
$ java -Xmx6g -jar target/r5.jar point --graphs ../dc-latest
```

GTFS/OSM data can be found here: https://s3.amazonaws.com/gtfs-bundles/dc-latest.zip

**Comment by buma** [comment]
I didn't do any specific speed optimizations yet. But 20 seconds seems too long.

I'll look into it.

BTW how did you manage to build graphs at all? I got duplicateFeedId problem: conveyal/gtfs-lib/issues/21.

**Comment by trevorgerhardt** [comment]
Thanks. We really can't proceed with live testing at this speed.

I might be using an out of date version of GTFS lib, haven't re-mvn
installed it in a while.

**Comment by buma** [comment]
For bike renting you also need this file in graph directory: http://www.capitalbikeshare.com/data/stations/bikeStations.xml
and build-config:
```json
{
        "bikeRentalFile" : "bikeStations.xml"
}
```

I checked and most of the time is spent in street search. Transit search itself takes 220ms but street searches takes 14-15 seconds. This is because currently we do all the street searches sequentially. 
Your request has at least 14 street searches (bike sharing needs 3 P+R needs 2 the other modes needs 1) and a couple more for searching street transfers between transits but those are in ms range.

Some can be run in parallel. Maybe some can be also reused. (For example access walk search with 2km limit to find stops then limit is raised for direct mode and search is made on the same graph)

I lowered run time from 20 to 5 seconds when I reduced distance limit in street search. (Currently it is 100km for direct search [and 2km for access/egress]  which takes the most time and I reduced it to 40 minutes) I also noticed that there were some changes in street search by other people made and it is quite strange now. I'm also not sure if distanceLimits are actually used since there were some changes in dominance functions. For example bicycle search prefers walking in wrong direction on streets instead of cycling on cycleway for some reason.

Currently maxBike and other max/min options in GraphQL request aren't used at all I'll change the query so those values will be used and runtime should be much better.
